Including "boot/picobin.h" triggers include ordering error it caused itself

Hey guys, this one just needs a one-line fix somewhere:

#### Problem

When using 
```c
#include "boot/picobin.h"
```

before any other includes, this compilation error is emitted:
```
/usr/local/pico-sdk/src/rp2350/pico_platform/include/pico/platform.h:20:2: error: #error pico/platform.h should not be included directly; include pico.h instead
   20 | #error pico/platform.h should not be included directly; include pico.h instead
```

This is initially confusing, however, because the user obviously didn't try to include `pico/platform.h` directly. Furthermore, including `pico.h` directly instead (the solution indicated) does not actually cause the inclusion of `boot/picobin.h`.

#### Expected Outcome

If `boot/picobin.h` is intended to be included directly by the user, it should include its own dependencies. (As it's included directly [here](https://github.com/raspberrypi/pico-examples/blob/4c3a3dc0196dd426fddd709616d0da984e027bab/flash/partition_info/partition_info.c#L10) I'm assuming that's the case.) Otherwise, it should emit an error identifying a header that can be included that includes `boot/picobin.h`, after including its dependencies.

--- a/src/common/boot_picobin_headers/include/boot/picobin.h
+++ b/src/common/boot_picobin_headers/include/boot/picobin.h
@@ -7,15 +7,11 @@
 #ifndef _BOOT_PICOBIN_H
 #define _BOOT_PICOBIN_H
 
-#ifndef NO_PICO_PLATFORM
-#include "pico/platform.h"
-#else
 #ifndef _u
 #ifdef __ASSEMBLER__
 #define _u(x) x
 #else
 #define _u(x) x ## u
-#endif
 #endif
 #endif
 
@@ -149,6 +145,7 @@
 #ifndef __ASSEMBLER__
 
 #include <stdbool.h>
+#include <stdint.h>
 
 typedef struct {
     // these must all be word aligned

